installation/routines/setup_git.sh

```diff
@@ -1,15 +1,88 @@
-#!/usr/bin/env bash
+convert_tardir_git_repo() {
+  echo "****************************************************"
+  echo "*** Converting tar-ball download into git repository"
+  echo "****************************************************"
+
+  git init -b "${GIT_BRANCH}"
+  git config pull.rebase false
+
+  # We always add origin as the selected (possible) user repository
+  # and MiczFlor's repository as upstream
+  # This means for developers everything is fully set up.
+  # For users there is no difference (origin = upstream = MiczFlor)
+  # We need to get the branch with larger depth, as we do not know
+  # how many commits happened between download and git repo init
+  if [[ $GIT_USE_SSH = true ]]; then
+      git remote add origin "git@github.com:${GIT_USER}/${GIT_REPO_NAME}.git"
+      git remote add upstream "git@github.com:MiczFlor/${GIT_REPO_NAME}.git"
+    if [[ $(git fetch origin "${GIT_BRANCH}") -ne 0 ]]; then
+      echo "*** Git fetch *************************************"
+      echo "Error in getting Git Repository using SSH!"
+      echo "Did you forget to upload the ssh key for this machine to GitHub?"
+      echo "Defaulting to HTTPS protocol. You can change back to SSH later with"
+      echo "git remote set-url origin git@github.com:${GIT_USER}/${GIT_REPO_NAME}.git"
+      echo "git remote set-url upstream git@github.com:MiczFlor/${GIT_REPO_NAME}.git"
+      echo "*** Git remotes ***********************************"
+      GIT_USE_SSH=false
+    fi
+  fi
+
+  if [[ $GIT_USE_SSH = false ]]; then
+      git remote add origin "https://github.com/${GIT_USER}/${GIT_REPO_NAME}.git"
+      git remote add upstream "https://github.com/MiczFlor/${GIT_REPO_NAME}.git"
+      git fetch origin "${GIT_BRANCH}"
+  fi
+
+  # In case we get a non-develop or non-main branch, we speculatively
+  # try to get these branches, so they can be checkout out with
+  # git checkout future3/develop
+  # without the need to set up the remote tracking information
+  # However, in a user repository, these may not be present, so we suppress output in these cases
+  if [[ $GIT_BRANCH != "future3/main" ]]; then
```

Comment:
Same here .. those values might be different in the future

Reply:
Done.